packages/clay-form/src/Input.tsx

```diff
@@ -155,6 +156,97 @@ const GroupInsetItem = React.forwardRef<
 
 GroupInsetItem.displayName = 'ClayInputGroupInsetItem';
 
+interface IInlineTextProps
+	extends React.HTMLAttributes<
+		HTMLDivElement | HTMLSpanElement | HTMLLabelElement
+	> {
+	/**
+	 * The id of the component
+	 */
+	id?: string;
+
+	/**
+	 * Text that hints at the expected data type to be entered.
+	 */
+	placeholder?: string;
+
+	/**
+	 * A flag that sets the component as noneditable.
+	 */
+	readOnly?: boolean;
+
+	/**
+	 * Path to the spritemap that Icon should use when referencing symbols.
+	 */
+	spritemap?: string;
+}
+
+const InlineText = React.forwardRef<
+	HTMLDivElement | HTMLSpanElement | HTMLLabelElement,
+	IInlineTextProps
+>(
+	({
+		children,
+		className,
+		id,
+		placeholder,
+		readOnly,
+		spritemap,
+		...otherProps
+	}: IInlineTextProps) => {
+		const inputRef = React.useRef<HTMLDivElement>(null);
+
+		const [inputValue, setInputValue] = React.useState('');
```

Comment:
To follow the pattern of other components we need the state to be controlled or uncontrolled you can see how we implemented this in Autocomplete for example.